tools/c7n_azure/tests_azure/tests_resources/test_postgresql_server.py

```diff
@@ -16,6 +18,12 @@
 AZURE_CLOUD_IPS = ['0.0.0.0']
 
 
+class AttrDict(dict):
+    def __init__(self, *args, **kwargs):
+        super(AttrDict, self).__init__(*args, **kwargs)
+        self.__dict__ = self
```

Comment:
what's the intent on this? the definition is quite odd.

Reply:
It's a simple way to have arbitrary data in a class that can be accessed both as attributes and using `[]`. I use it to mock return values where attributes are needed. `Mock` in this case makes comparisons more difficult as it can't be directly compared.

Reply:
fwiw, we have one of these already in c7n.config.Bag

Reply:
It doesn't look like this is being used anywhere in this diff, can we remove this?

Reply:
Yep, forgot to remove it when I refactored - removed now.